plostd.c

```diff
@@ -293,7 +293,8 @@ void plostd_printf(char attr, const char *fmt, ...)
 			break;
 		case 'x':
 			i = va_arg(ap, int);
-			plostd_puts(plostd_itoah((u8 *)&i, 2, buff, 0));
+			plostd_puts("0x");
```

Comment:
`printf("%x", 0xab12)` never prints `0xab12`, but `ab12`, maybe You thought about  `case 'p'` for the pointers which indeed prints `0x` before the hex address.

Reply:
Yes, you have right, I've made mistake.
I've already changed it to print `0x` in `case 'p'`.